web-common/src/features/explores/selectors.ts

```diff
@@ -73,6 +76,27 @@ export function useExploreValidSpec(
   );
 }
 
+export function getExploreValidSpecOptions(exploreNameStore: Readable<string>) {
+  return derived([runtime, exploreNameStore], ([{ instanceId }, exploreName]) =>
+    getRuntimeServiceGetExploreQueryOptions(
+      instanceId,
+      {
+        name: exploreName,
+      },
+      {
+        query: {
+          select: (data) =>
+            <ExploreValidSpecResponse>{
+              explore: data.explore?.explore?.state?.validSpec,
+              metricsView: data.metricsView?.metricsView?.state?.validSpec,
```

Comment:
Nit: I think it'd be easier to read the consumers if these variable names were `exploreSpec` (or `exploreValidSpec`) and `metricsViewSpec` (or `metricsViewValidSpec`). I was initially confused when I saw that this `explore` variable was a `V1ExploreSpec` type.

Reply:
Ya makes sense. Lets keep it `metricsViewSpec` and `exploreSpec`. Valid spec is in method name.